Source/Scene/Cesium3DTileGroundPrimitive.js

```diff
@@ -0,0 +1,875 @@
+
+                var normal = ellipsoid.geodeticSurfaceNormal(position, scratchNormal);
+                var scaledPosition = ellipsoid.scaleToGeodeticSurface(position, position);
+                var scaledNormal = Cartesian3.multiplyByScalar(normal, minHeight, scratchScaledNormal);
+                var minHeightPosition = Cartesian3.add(scaledPosition, scaledNormal, scratchMinHeightPosition);
+
+                scaledNormal = Cartesian3.multiplyByScalar(normal, maxHeight, scaledNormal);
+                var maxHeightPosition = Cartesian3.add(scaledPosition, scaledNormal, scratchMaxHeightPosition);
+
+                Cartesian3.subtract(maxHeightPosition, center, maxHeightPosition);
+                Cartesian3.subtract(minHeightPosition, center, minHeightPosition);
+
+                Cartesian3.pack(maxHeightPosition, batchedPositions, positionIndex);
+                Cartesian3.pack(minHeightPosition, batchedPositions, positionIndex + 3);
+
+                batchedIds[idIndex] = i;
+                batchedIds[idIndex + 1] = i;
+
+                positionIndex += 6;
+                colorIndex += 8;
+                idIndex += 2;
+            }
+
+            var rectangle = scratchBVRectangle;
+            rectangle.west = minLon;
+            rectangle.east = maxLon;
+            rectangle.south = minLat;
+            rectangle.north = maxLat;
+
+            boundingVolumes[i] = OrientedBoundingBox.fromRectangle(rectangle, minHeight, maxHeight, ellipsoid);
+
+            var indicesIndex = object.indexOffset;
+
+            var indexOffset = indexOffsets[i];
+            var indexCount = indexCounts[i];
+
+            batchedIndexOffsets[i] = indicesIndex;
+
+            for (j = 0; j < indexCount; j += 3) {
+                var i0 = indices[indexOffset + j] - polygonOffset;
+                var i1 = indices[indexOffset + j + 1] - polygonOffset;
+                var i2 = indices[indexOffset + j + 2] - polygonOffset;
+
+                batchedIndices[indicesIndex++] = i0 * 2 + positionOffset;
+                batchedIndices[indicesIndex++] = i1 * 2 + positionOffset;
+                batchedIndices[indicesIndex++] = i2 * 2 + positionOffset;
+
+                batchedIndices[indicesIndex++] = i2 * 2 + 1 + positionOffset;
+                batchedIndices[indicesIndex++] = i1 * 2 + 1 + positionOffset;
+                batchedIndices[indicesIndex++] = i0 * 2 + 1 + positionOffset;
+            }
+
+            for (j = 0; j < polygonCount - 1; ++j) {
+                batchedIndices[indicesIndex++] = j * 2 + 1 + positionOffset;
+                batchedIndices[indicesIndex++] = (j + 1) * 2 + positionOffset;
+                batchedIndices[indicesIndex++] = j * 2 + positionOffset;
+
+                batchedIndices[indicesIndex++] = j * 2 + 1 + positionOffset;
+                batchedIndices[indicesIndex++] = (j + 1) * 2 + 1 + positionOffset;
+                batchedIndices[indicesIndex++] = (j + 1) * 2 + positionOffset;
+            }
```

Comment:
Similar to above, I think some comments would go a long way to explaining how the shape is being created.